packages/openapi-v3/test/unit/controller-spec/controller-decorators/param-decorators/param.test.ts

```diff
@@ -210,301 +187,56 @@ describe('Routing metadata for parameters', () => {
-
-      const pageSizeSpec: ParameterObject = {
-        name: 'pageSize',
-        type: 'number',
-        in: 'query',
-      };
-
-      class MyController {
-        @get('/')
-        @param(offsetSpec)
-        @param(pageSizeSpec)
-        list(offset?: number, pageSize?: number) {}
-      }
-
-      const actualSpec = getControllerSpec(MyController);
-
-      expect(actualSpec.paths['/']['get'].parameters).to.eql([
-        offsetSpec,
-        pageSizeSpec,
-      ]);
-    });
-
-    it('can define multiple parameters by arguments', () => {
-      const offsetSpec: ParameterObject = {
-        name: 'offset',
-        type: 'number',
-        in: 'query',
-      };
-
-      const pageSizeSpec: ParameterObject = {
-        name: 'pageSize',
-        type: 'number',
-        in: 'query',
-      };
-
-      class MyController {
-        @get('/')
-        list(
-          @param(offsetSpec) offset?: number,
-          @param(pageSizeSpec) pageSize?: number,
-        ) {}
-      }
-
-      const actualSpec = getControllerSpec(MyController);
-
-      expect(actualSpec.paths['/']['get'].parameters).to.eql([
-        offsetSpec,
-        pageSizeSpec,
-      ]);
-    });
-    // tslint:disable-next-line:max-line-length
-    it('throws an error if @param is used at both method and parameter level', () => {
-      expect(() => {
-        const offsetSpec: ParameterObject = {
-          name: 'offset',
-          type: 'number',
-          in: 'query',
-        };
-
-        const pageSizeSpec: ParameterObject = {
-          name: 'pageSize',
-          type: 'number',
-          in: 'query',
-        };
-        // tslint:disable-next-line:no-unused-variable
-        class MyController {
-          @get('/')
-          @param(offsetSpec)
-          list(offset?: number, @param(pageSizeSpec) pageSize?: number) {}
-        }
-      }).to.throw(
-        /Mixed usage of @param at method\/parameter level is not allowed/,
-      );
-    });
-
     it('adds to existing spec provided via @operation', () => {
       const offsetSpec: ParameterObject = {
         name: 'offset',
-        type: 'number',
         in: 'query',
+        schema: {
+          type: 'number',
+        },
       };
 
       const pageSizeSpec: ParameterObject = {
         name: 'pageSize',
-        type: 'number',
         in: 'query',
+        schema: {
+          type: 'number',
+        },
       };
 
       const responses: ResponsesObject = {
         200: {
-          schema: {
-            type: 'string',
+          content: {
+            '*/*': {
+              schema: {
+                type: 'string',
+              },
+            },
           },
           description: 'a string response',
         },
       };
 
       class MyController {
         @operation('get', '/', {responses})
-        @param(offsetSpec)
-        @param(pageSizeSpec)
-        list(offset?: number, pageSize?: number) {}
+        list(
+          @param({name: 'offset', in: 'query'})
```

Comment:
Since this is no longer using `offsetSpec` variable defined above, can we simplify it using `param.query`?

```ts
@param.query('offset') offset?: number
```

Same comment applies to `pageSize` below.